Troubleshooting guide (#99)
magicaks/magicaks: setup.md

* Azure Policy enabled on the cluster (no policies assigned right now)
* Azure Firewall integrated with network and application rules as recommended by AKS
* Grafana connected to Log Analytics workspace of the cluster is running in Azure Container Instances backed by managed PostgreSQL database

## Troubleshooting

> - Terraform auth error

Reviewer: ```suggestion
* Terraform authentication error
```

    ```bash
    Error: Failed to get existing workspaces: containers.Client#ListBlobs: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthenticationFailed" Message="Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:e4c5cf49-801e-0068-4539-0cb9e7000000\nTime:2021-02-26T12:18:40.6499706Z"
    ```

>     This is due to WSL clock skew. Fix it by running this command:

Reviewer: ```suggestion

    This is due to WSL clock skew. Fix it by running this command:

```

Reviewer 2: In addition - you may get this or a similar error if you have left wsl open over night and need to re-authenticate with az login

    ```bash
    sudo hwclock -s
    ```

> -  Line endings not correct for bash scripts.

Reviewer: ```suggestion
*  Line endings not correct for bash scripts
```

    ```bash
    Error: Error running command '/mnt/c//2-provision-aks/getcreds.sh ': exit status 127. Output: /bin/sh: 1: /mnt/c//2-provision-aks/getcreds.sh: not found

>     ```

Reviewer: ```suggestion
    ```

```

>     *We know it's the 21 century, but check the line endings*. Line endings must LF instead of CRLF (this may happen if you clone the repo from windows for example.) - check the line endings of all sh files in the repo at once and set them to LF.

Reviewer: ```suggestion
    *We know it's the 21 century, but check the line endings*. Line endings must be in LF format instead of CRLF. You may run into this issue, if you clone the repository in Windows for example. Check the line endings of all the bash script files (`*.sh`) in the repository and set them to LF format to correct the problem.
```

>     You can use the `dos2unix` utilty to do the conversion. Install it via apt-get.

Reviewer: ```suggestion
    You can use the `dos2unix` utility to do the conversion. Install it via `apt-get`.
```

> - When does this happen Tess?

Reviewer: ```suggestion
* When does this happen, Tess?
```

Reviewer 2: @tompaana I don't think this line should be in the repo :)

@lybecker - It would happen either if someone gits from windows (with CRLF as their default) or git clones on Windows

    ```bash
    Error: failed to create resource: namespaces "app1" not found

>     ```

Reviewer: ```suggestion
    ```

```

>     This is due to a timing issue where `app1` is not yet created - Wait a few minutes and rerun the apply, and it should work.

Reviewer: ```suggestion
    This is due to a timing issue where `app1` is not yet created - wait a few minutes, rerun the apply and it should work.
```

* Azure Policy enabled on the cluster (no policies assigned right now)
* Azure Firewall integrated with network and application rules as recommended by AKS
* Grafana connected to Log Analytics workspace of the cluster is running in Azure Container Instances backed by managed PostgreSQL database

## Troubleshooting

* Terraform authentication error.

    ```bash
    Error: Failed to get existing workspaces: containers.Client#ListBlobs: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthenticationFailed" Message="Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:e4c5cf49-801e-0068-4539-0cb9e7000000\nTime:2021-02-26T12:18:40.6499706Z"
    ```

    This is due to WSL clock skew. Fix it by running this command:

    ```bash
    sudo hwclock -s
    ```

> * Line endings not correct for bash scripts. E.g. by git clone/pull from Windows (with CRLF as their default).

Reviewer: ```suggestion
* Line endings not correct for bash scripts, e.g. by Git clone/pull from Windows (with CRLF as the default).
```
